api/organization_test.go

```diff
@@ -0,0 +1,40 @@
+package api_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/env0/terraform-provider-env0/api"
+)
+
+var _ = Describe("Organization", func() {
+	var client *ApiClient
+	var organization Organization
+
+	BeforeEach(func() {
+		var err error
+		client, err = NewClientFromEnv()
+		Expect(err).To(BeNil())
+		Expect(client).ToNot(BeNil())
+	})
+
+	JustBeforeEach(func() {
+		var err error
+		organization, err = client.Organization()
+		Expect(err).To(BeNil())
+	})
```

Comment:
```suggestion
	BeforeEach(func() {
		var err error
		client, err = NewClientFromEnv()
	})

	JustBeforeEach(func() {
		var err error
		organization, err = client.Organization()
	})
	
	It('should initialize the api client', func() {
		Expect(err).To(BeNil())
		Expect(client).ToNot(BeNil())
        })
        
	It('should set the organization', func() {
  	        Expect(err).To(BeNil())
	})
```

Reply:
i have an issue with writing the test like this: the order of the assertion causes the feedback from the test to be very noisy:
if client is nil, then you just get a whole lot of panic stack traces printed, and it's unclear what went wrong. if organization was a pointer, not a struct, it's even worse. since it is a struct, then just everything fails, so it's still way more noise than i would expect. So a simple thing like forgetting the test depends on ENV0_API_KEY can really blow up the screen. here is the code i used, tried to keep it as close to your suggestion as possible (added one line to emulate NewClientFromEnv failing):

```go
var _ = Describe("Organization", func() {
	var client *ApiClient
	var clientErr error
	var organization Organization
	var organizationErr error

	BeforeEach(func() {
		client, clientErr = NewClientFromEnv()
		client = nil // <--- this emulates the error
	})

	JustBeforeEach(func() {
		organization, organizationErr = client.Organization()
	})

	Describe("Fetch organization data", func() {
		When("Fetching the default organization of given api key", func() {
			It("should initialize the api client", func() {
				Expect(clientErr).To(BeNil())
				Expect(client).ToNot(BeNil())
			})
			It("should set the organization", func() {
				Expect(organizationErr).To(BeNil())
	  		})
			It("Should have id set", func() {
				Expect(organization.Id).ToNot(BeEmpty())
			})
			It("Should have name set", func() {
				Expect(organization.Name).ToNot(BeEmpty())
			})
			It("Should not be self hosted", func() {
				Expect(organization.IsSelfHosted).To(BeFalse())
			})
		})
	})
})
```

and here is the printout:
```
Running Suite: Api Suite
========================
Random Seed: 1616569563
Will run 4 of 4 specs

•! Panic in Spec Setup (JustBeforeEach) [0.001 seconds]
Organization
/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:10
  Fetch organization data [JustBeforeEach]
  /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:26
    when Fetching the default organization of given api key
    /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:27
      should initialize the api client
      /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:28

      Test Panicked
      runtime error: invalid memory address or nil pointer dereference
      /usr/local/go/src/runtime/panic.go:212

      Full Stack Trace
      github.com/env0/terraform-provider-env0/api.(*ApiClient).normalizeEndpoint(0x0)
      	/home/shlomi/env0/terraform-provider-env0/api/api_client.go:37 +0x2c
      github.com/env0/terraform-provider-env0/api.(*ApiClient).request(0x0, 0x523)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:10 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).getJSON(0x0, 0x88a00f, 0xe, 0x0, 0x7d5f60, 0xc00000ed80, 0x0, 0x3)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:41 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).Organization(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/env0/terraform-provider-env0/api/organization.go:9 +0x9e
      github.com/env0/terraform-provider-env0/api_test.glob..func1.2()
      	/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:22 +0x7a
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc000072300, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:113 +0xa3
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc000072300, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:64 +0xcf
      github.com/onsi/ginkgo/internal/leafnodes.(*SetupNode).Run(0xc00048a0d0, 0x9283c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/setup_nodes.go:15 +0x64
      github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc0007bc000, 0x0, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:205 +0x4c7
      github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc0007bc000, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:138 +0xf2
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc0000dd4a0, 0xc0007bc000, 0x0)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:200 +0x10f
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc0000dd4a0, 0x1)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:170 +0x120
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc0000dd4a0, 0xc0001d2e50)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:66 +0x117
      github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc000105490, 0x7f5a62ed95f8, 0xc00079c480, 0x886cd6, 0x9, 0xc0002daa10, 0x1, 0x1, 0x9312a0, 0xc0000d8a80, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/suite/suite.go:79 +0x586
      github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0xc0005d7728, 0x1, 0x1, 0x40e808)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:229 +0x217
      github.com/onsi/ginkgo.RunSpecs(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0x48ec37965c)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:210 +0x16e
      github.com/env0/terraform-provider-env0/api_test.TestApi(0xc00079c480)
      	/home/shlomi/env0/terraform-provider-env0/api/api_suite_test.go:12 +0x96
      testing.tRunner(0xc00079c480, 0x8b0370)
      	/usr/local/go/src/testing/testing.go:991 +0xdc
      created by testing.(*T).Run
      	/usr/local/go/src/testing/testing.go:1042 +0x357
------------------------------
•! Panic in Spec Setup (JustBeforeEach) [0.000 seconds]
Organization
/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:10
  Fetch organization data [JustBeforeEach]
  /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:26
    when Fetching the default organization of given api key
    /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:27
      Should have id set
      /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:32

      Test Panicked
      runtime error: invalid memory address or nil pointer dereference
      /usr/local/go/src/runtime/panic.go:212

      Full Stack Trace
      github.com/env0/terraform-provider-env0/api.(*ApiClient).normalizeEndpoint(0x0)
      	/home/shlomi/env0/terraform-provider-env0/api/api_client.go:37 +0x2c
      github.com/env0/terraform-provider-env0/api.(*ApiClient).request(0x0, 0x523)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:10 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).getJSON(0x0, 0x88a00f, 0xe, 0x0, 0x7d5f60, 0xc00000ee60, 0x0, 0x3)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:41 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).Organization(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/env0/terraform-provider-env0/api/organization.go:9 +0x9e
      github.com/env0/terraform-provider-env0/api_test.glob..func1.2()
      	/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:22 +0x7a
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc000072300, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:113 +0xa3
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc000072300, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:64 +0xcf
      github.com/onsi/ginkgo/internal/leafnodes.(*SetupNode).Run(0xc00048a0d0, 0x9283c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/setup_nodes.go:15 +0x64
      github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc0007bc0f0, 0x0, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:205 +0x4c7
      github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc0007bc0f0, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:138 +0xf2
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc0000dd4a0, 0xc0007bc0f0, 0x1)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:200 +0x10f
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc0000dd4a0, 0x1)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:170 +0x120
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc0000dd4a0, 0xc0001d2e50)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:66 +0x117
      github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc000105490, 0x7f5a62ed95f8, 0xc00079c480, 0x886cd6, 0x9, 0xc0002daa10, 0x1, 0x1, 0x9312a0, 0xc0000d8a80, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/suite/suite.go:79 +0x586
      github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0xc0005d7728, 0x1, 0x1, 0x40e808)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:229 +0x217
      github.com/onsi/ginkgo.RunSpecs(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0x48ec37965c)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:210 +0x16e
      github.com/env0/terraform-provider-env0/api_test.TestApi(0xc00079c480)
      	/home/shlomi/env0/terraform-provider-env0/api/api_suite_test.go:12 +0x96
      testing.tRunner(0xc00079c480, 0x8b0370)
      	/usr/local/go/src/testing/testing.go:991 +0xdc
      created by testing.(*T).Run
      	/usr/local/go/src/testing/testing.go:1042 +0x357
------------------------------
•! Panic in Spec Setup (JustBeforeEach) [0.000 seconds]
Organization
/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:10
  Fetch organization data [JustBeforeEach]
  /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:26
    when Fetching the default organization of given api key
    /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:27
      Should have name set
      /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:35

      Test Panicked
      runtime error: invalid memory address or nil pointer dereference
      /usr/local/go/src/runtime/panic.go:212

      Full Stack Trace
      github.com/env0/terraform-provider-env0/api.(*ApiClient).normalizeEndpoint(0x0)
      	/home/shlomi/env0/terraform-provider-env0/api/api_client.go:37 +0x2c
      github.com/env0/terraform-provider-env0/api.(*ApiClient).request(0x0, 0x523)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:10 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).getJSON(0x0, 0x88a00f, 0xe, 0x0, 0x7d5f60, 0xc00000ef60, 0x0, 0x3)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:41 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).Organization(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/env0/terraform-provider-env0/api/organization.go:9 +0x9e
      github.com/env0/terraform-provider-env0/api_test.glob..func1.2()
      	/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:22 +0x7a
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc000072300, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:113 +0xa3
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc000072300, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:64 +0xcf
      github.com/onsi/ginkgo/internal/leafnodes.(*SetupNode).Run(0xc00048a0d0, 0x9283c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/setup_nodes.go:15 +0x64
      github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc0007bc1e0, 0x0, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:205 +0x4c7
      github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc0007bc1e0, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:138 +0xf2
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc0000dd4a0, 0xc0007bc1e0, 0x2)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:200 +0x10f
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc0000dd4a0, 0x1)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:170 +0x120
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc0000dd4a0, 0xc0001d2e50)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:66 +0x117
      github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc000105490, 0x7f5a62ed95f8, 0xc00079c480, 0x886cd6, 0x9, 0xc0002daa10, 0x1, 0x1, 0x9312a0, 0xc0000d8a80, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/suite/suite.go:79 +0x586
      github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0xc0005d7728, 0x1, 0x1, 0x40e808)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:229 +0x217
      github.com/onsi/ginkgo.RunSpecs(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0x48ec37965c)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:210 +0x16e
      github.com/env0/terraform-provider-env0/api_test.TestApi(0xc00079c480)
      	/home/shlomi/env0/terraform-provider-env0/api/api_suite_test.go:12 +0x96
      testing.tRunner(0xc00079c480, 0x8b0370)
      	/usr/local/go/src/testing/testing.go:991 +0xdc
      created by testing.(*T).Run
      	/usr/local/go/src/testing/testing.go:1042 +0x357
------------------------------
•! Panic in Spec Setup (JustBeforeEach) [0.000 seconds]
Organization
/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:10
  Fetch organization data [JustBeforeEach]
  /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:26
    when Fetching the default organization of given api key
    /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:27
      Should not be self hosted
      /home/shlomi/env0/terraform-provider-env0/api/organization_test.go:38

      Test Panicked
      runtime error: invalid memory address or nil pointer dereference
      /usr/local/go/src/runtime/panic.go:212

      Full Stack Trace
      github.com/env0/terraform-provider-env0/api.(*ApiClient).normalizeEndpoint(0x0)
      	/home/shlomi/env0/terraform-provider-env0/api/api_client.go:37 +0x2c
      github.com/env0/terraform-provider-env0/api.(*ApiClient).request(0x0, 0x523)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:10 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).getJSON(0x0, 0x88a00f, 0xe, 0x0, 0x7d5f60, 0xc00000f080, 0x0, 0x3)
      	/home/shlomi/env0/terraform-provider-env0/api/http_verbs_with_json.go:41 +0x2f
      github.com/env0/terraform-provider-env0/api.(*ApiClient).Organization(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/env0/terraform-provider-env0/api/organization.go:9 +0x9e
      github.com/env0/terraform-provider-env0/api_test.glob..func1.2()
      	/home/shlomi/env0/terraform-provider-env0/api/organization_test.go:22 +0x7a
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc000072300, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:113 +0xa3
      github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc000072300, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/runner.go:64 +0xcf
      github.com/onsi/ginkgo/internal/leafnodes.(*SetupNode).Run(0xc00048a0d0, 0x9283c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/leafnodes/setup_nodes.go:15 +0x64
      github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc0007bc2d0, 0x0, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:205 +0x4c7
      github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc0007bc2d0, 0x9283c0, 0xc0000d8a80)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/spec/spec.go:138 +0xf2
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc0000dd4a0, 0xc0007bc2d0, 0x0)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:200 +0x10f
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc0000dd4a0, 0x1)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:170 +0x120
      github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc0000dd4a0, 0xc0001d2e50)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/specrunner/spec_runner.go:66 +0x117
      github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc000105490, 0x7f5a62ed95f8, 0xc00079c480, 0x886cd6, 0x9, 0xc0002daa10, 0x1, 0x1, 0x9312a0, 0xc0000d8a80, ...)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/internal/suite/suite.go:79 +0x586
      github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0xc0005d7728, 0x1, 0x1, 0x40e808)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:229 +0x217
      github.com/onsi/ginkgo.RunSpecs(0x928a40, 0xc00079c480, 0x886cd6, 0x9, 0x48ec37965c)
      	/home/shlomi/go/pkg/mod/github.com/onsi/ginkgo@v1.15.2/ginkgo_dsl.go:210 +0x16e
      github.com/env0/terraform-provider-env0/api_test.TestApi(0xc00079c480)
      	/home/shlomi/env0/terraform-provider-env0/api/api_suite_test.go:12 +0x96
      testing.tRunner(0xc00079c480, 0x8b0370)
      	/usr/local/go/src/testing/testing.go:991 +0xdc
      created by testing.(*T).Run
      	/usr/local/go/src/testing/testing.go:1042 +0x357
------------------------------


Summarizing 4 Failures:

[Panic!] Organization [JustBeforeEach] Fetch organization data when Fetching the default organization of given api key should initialize the api client 
/usr/local/go/src/runtime/panic.go:212

[Panic!] Organization [JustBeforeEach] Fetch organization data when Fetching the default organization of given api key Should have id set 
/usr/local/go/src/runtime/panic.go:212

[Panic!] Organization [JustBeforeEach] Fetch organization data when Fetching the default organization of given api key Should have name set 
/usr/local/go/src/runtime/panic.go:212

[Panic!] Organization [JustBeforeEach] Fetch organization data when Fetching the default organization of given api key Should not be self hosted 
/usr/local/go/src/runtime/panic.go:212

Ran 4 of 4 Specs in 0.004 seconds
FAIL! -- 0 Passed | 4 Failed | 0 Pending | 0 Skipped
--- FAIL: TestApi (0.01s)
FAIL
exit status 1
FAIL	github.com/env0/terraform-provider-env0/api	16.143s
```

so for me some expects are dependent - and err != nil should prevent spillage of errors on screen - just show the err that happens. If you still insist i'll do it this way, but IMHO some other solution should be considered.

Reply:
also, can you please suggest something regarding my original question? it was for project_test, not organization_test.

Reply:
Granted. If this particular prerequisite assertion will fail the output will be noisy.  
I think we can live with that and the general specification structure is much clearer to the reader.  

Regarding `project_test` - 
I think what you wanna do is assign the ids returned from the when part, then use those ids in subsequent `it`s.  
I hope I was able to understand what you meant and that my answer is clear - otherwise I can try and come up with an example